File: mime-type.py
import sys
import math

n = int(input())  # Number of elements which make up the association table.
q = int(input())  # Number Q of file names to be analyzed.
mime_dict = {}
unknown = "UNKNOWN"

for i in range(n):
  # ext: file extension
  # mt: MIME type.
  ext, mt = input().split()
  mime_dict[ext.lower()] = mt

for i in range(q):
  fname = input().lower()  # One file name per line.
  if fname.find(".") == -1:
    print(unknown)
    continue
    
  ext = fname.split(".").pop()
  if ext in mime_dict.keys():
    print(mime_dict[ext])
  else:
    print(unknown)

File: mime-type.swift
import Glibc
import Foundation

public struct StderrOutputStream: TextOutputStream {
  public mutating func write(_ string: String) { fputs(string, stderr) }
}
public var errStream = StderrOutputStream()

let n = Int(readLine()!)! // Number of elements which make up the association table.
let q = Int(readLine()!)! // Number Q of file names to be analyzed.
var mimeDict = [String: String]()
if n > 0 {
  for _ in 0...(n-1) {
    let inputs = (readLine()!).split(separator: " ").map(String.init)
    let ext = inputs[0].lowercased() // file extension
    let mt = inputs[1] // MIME type.
    mimeDict[ext] = mt
  }
}
if q > 0 {
  let unknown = "UNKNOWN"
  for _ in 0...(q-1) {
    let fname = readLine()! // One file name per line.
    let mime = fname.lowercased()
    if !mime.contains(".") {
      print(unknown)
      continue
    }
    let mime2 = mime.split(separator: ".", omittingEmptySubsequences: false)
                    .last
                    .map(String.init)
                    .flatMap { mimeDict[$0] }
    print(mime2 ?? unknown)
  }
}
